src/darwin/Framework/CHIP/MTRControllerFactory.mm

```diff
@@ -387,6 +389,15 @@ - (MTRDeviceController * _Nullable)startControllerOnExistingFabric:(MTRDeviceCon
         return nil;
     }
 
+    if (_otaProviderDelegateBridge) {
+        BOOL ok = [self maybeInitializeOTAProvider:_otaProviderDelegateBridge];
+        if (ok == NO) {
+            MTR_LOG_ERROR("Failed to init provider delegate bridge");
+            [self controllerShuttingDown:controller];
```

Comment:
No, at this point the controller has started up, so we need to actually call `shutdown` on it so it will shut itself down properly.  `controllerShuttingDown` will only do part of that work.
```suggestion
            [controller shutdown];
```